diff/erofs/erofs.go

```diff
@@ -0,0 +1,180 @@
+			config := ic.Config.(*Config)
+			return NewErofsDiff(cs, config.MediaHandlers)
+		},
+	})
+}
+
+// CompareApplier handles both comparison and
+// application of layer diffs.
+type CompareApplier interface {
+	diff.Applier
+	diff.Comparer
+}
+
+// erofsDiff does erofs comparison and application
+type erofsDiff struct {
+	store         content.Store
+	globalHandler map[string]MediaHandler
+}
+
+var emptyDesc = ocispec.Descriptor{}
+
+// NewErofsDiff is the erofs container layer implementation
+func NewErofsDiff(store content.Store, handlers map[string]MediaHandler) (CompareApplier, error) {
+	return erofsDiff{
+		store:         store,
+		globalHandler: handlers,
+	}, nil
+}
+
+// Apply applies the content associated with the provided digests onto the
+// provided mounts. Archive content will be extracted and decompressed if
+// necessary.
+func (s erofsDiff) Apply(ctx context.Context, desc ocispec.Descriptor, mounts []mount.Mount, opts ...diff.ApplyOpt) (d ocispec.Descriptor, err error) {
+	t1 := time.Now()
+	defer func() {
+		if err == nil {
+			log.G(ctx).WithFields(logrus.Fields{
+				"d":      time.Since(t1),
+				"digest": desc.Digest,
+				"size":   desc.Size,
+				"media":  desc.MediaType,
+			}).Debugf("diff applied")
+		}
+	}()
+
+	mediaHandler, ok := s.globalHandler[desc.MediaType]
+	if !ok {
+		return emptyDesc, fmt.Errorf("currently unsupported media type: %s", desc.MediaType)
+	}
+
+	var config diff.ApplyConfig
+	for _, o := range opts {
+		if err := o(ctx, desc, &config); err != nil {
+			return emptyDesc, fmt.Errorf("failed to apply config opt: %w", err)
+		}
+	}
+
+	ra, err := s.store.ReaderAt(ctx, desc)
+	if err != nil {
+		return emptyDesc, fmt.Errorf("failed to get reader from content store: %w", err)
+	}
+	defer ra.Close()
+
+	rc := &readCounter{
+		r: content.NewReader(ra),
+	}
+
+	if err := apply(ctx, mounts, rc, mediaHandler); err != nil {
+		return emptyDesc, err
+	}
+
+	log.G(ctx).Infof("read %v from content", rc.c)
+	// Read any trailing data
+	if _, err := io.Copy(io.Discard, rc); err != nil {
+		return emptyDesc, err
+	}
+
+	return ocispec.Descriptor{
+		MediaType: ocispec.MediaTypeImageLayer,
+		Size:      rc.c,
+		Digest:    desc.Digest,
+	}, nil
+}
+
+// Compare creates a diff between the given mounts and uploads the result
+// to the content store.
+func (s erofsDiff) Compare(ctx context.Context, lower, upper []mount.Mount, opts ...diff.Opt) (d ocispec.Descriptor, err error) {
+	return emptyDesc, fmt.Errorf("erofsDiff does not implement Compare method: %w", errdefs.ErrNotImplemented)
+}
+
+func apply(ctx context.Context, mounts []mount.Mount, r io.Reader, handler MediaHandler) error {
+	return mount.WithTempMount(ctx, mounts, func(root string) error {
+		cmd := exec.CommandContext(ctx, handler.BinaryName, handler.Args...)
+		cmd.Args = append(cmd.Args, filepath.Join(root, "layer.erofs"))
+		cmd.Stdin = r
+		out, err := cmd.CombinedOutput()
+		if err != nil {
+			return err
+		}
```

Comment:
What does it output? Just in case, I'd like to include `out` in the returning error in case its stdout has something interesting.